CONFIGURATION.md

```diff
@@ -1,79 +1,171 @@
-# Configuration
+# SolarWinds APM Ruby Configuration Guide
 
-By default all applicable instrumentations are enabled. The only required configuration is the service key, so a minimal example to get started is:
+This guide covers all configuration options for the SolarWinds APM Ruby gem, an OpenTelemetry-based distribution that provides automatic instrumentation and observability features for Ruby applications.
+
+## Table of Contents
+
+- [Quick Start](#quick-start)
+- [Configuration Precedence](#configuration-precedence)
+- [Environment Variables](#environment-variables)
+- [Configuration Files](#configuration-files)
+- [Programmatic Configuration](#programmatic-configuration)
+- [Advanced Configuration](#advanced-configuration)
+- [Configuration Reference](#configuration-reference)
+- [Troubleshooting](#troubleshooting)
+
+## Quick Start
+
+To get started quickly, you only need to set your service key:
 
 ```bash
-export SW_APM_SERVICE_KEY=<set-service-key-here>
+export SW_APM_SERVICE_KEY=<your-api-token>:<your-service-name>
 ```
 
-Configuration can be set several ways, with the following precedence:
+By default, all applicable instrumentations are enabled and the gem works out-of-the-box with sensible defaults.
+
+### Minimal Example
+
+```ruby
+# Set the service key (required)
+ENV['SW_APM_SERVICE_KEY'] = 'your-api-token:my-ruby-app'
+
+# Require the gem (typically done automatically by Bundler)
+require 'solarwinds_apm'
+
+# Your application code here
+```
+
+## Configuration Precedence
+
+Configuration can be set in multiple ways with the following precedence order (highest to lowest):
+
+1. **Environment Variables** - Highest priority
+2. **Programmatic Configuration** - Set in Ruby code
+3. **Configuration Files** - Rails initializer or config file
+4. **Default Values** - Built-in defaults
+
+> **💡 Tip:** Environment variables always take precedence, making them ideal for deployment-specific settings.
 
-`environment variable > programmatic > configuration file > default`
+Environment variables are the most flexible way to configure the SolarWinds APM gem, especially in containerized or cloud environments.
 
-## Environment Variables
+All SolarWinds APM-specific settings are prefixed with `SW_APM_`. Standard OpenTelemetry environment variables are also supported where applicable.
```

Comment:
If there is a central upstream document that lists the available OTel env vars, then that would be good to link here:

```suggestion
All SolarWinds APM-specific settings are prefixed with `SW_APM_`. [Standard OpenTelemetry environment variables](https://link-to-doc) are also supported where applicable.
```